quadbin/main.py

```diff
@@ -0,0 +1,534 @@
+from .utils import (
+    clip_latitude,
+    clip_longitude,
+    point_to_tile,
+    tile_to_longitude,
+    tile_to_latitude,
+)
+
+
+def is_valid_index(index):
+    """Return True if this is a valid Quadbin index.
+
+    Parameters
+    ----------
+    index : int
+
+    Returns
+    -------
+    bool
+    """
+    header = 0x4000000000000000
+    mode = (index >> 59) & 7
+    resolution = (index >> 52) & 0x1F
+    unused = 0xFFFFFFFFFFFFF >> (resolution << 2)
+    return (
+        index >= 0
+        and (index & header == header)
+        and mode in [0, 1, 2, 3, 4, 5, 6]
+        and resolution >= 0
+        and resolution <= 26
+        and (index & unused == unused)
+    )
+
+
+def is_valid_cell(cell):
+    """Return True if this is a valid Quadbin cell (mode 1).
+
+    Parameters
+    ----------
+    index : int
+
+    Returns
+    -------
+    bool
+    """
+    header = 0x4000000000000000
+    mode = (cell >> 59) & 7
+    resolution = (cell >> 52) & 0x1F
+    unused = 0xFFFFFFFFFFFFF >> (resolution << 2)
+    return (
+        cell >= 0
+        and (cell & header == header)
+        and mode == 1
+        and resolution >= 0
+        and resolution <= 26
+        and (cell & unused == unused)
+    )
+
+
+def cell_to_tile(cell):
+    """Convert a cell into a tile.
+
+    Parameters
+    ----------
+    cell : int
+
+    Returns
+    -------
+    tile: tuple (x, y, z)
+    """
+    b = [
```

Comment:
Probably it's better for maintainability (and reusability) to move some function variables as "constants" to the upper side of the module